### PR TITLE
feat: add sum onnx import

### PR DESCRIPTION
**Pull request by JachymPutta**
### Checklist
- [X] Confirmed that `run-checks all` script has been executed.
- [X] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

Helping with #1714 

### Changes

Added the ONNX models for sum and sum_int, the node seems to by covered by `add`

### Testing

Manual

`cargo test -p burn-import --color=always -- --color=always`
`cargo test -p onnx-tests --color=always`

### CODE REVIEW

**Comment by antimora** [comment]
Yeah I think Add and Sum are similar but not the same. The python script generates Add node types. Probably it's difficult or not possible to achieve with PyTorch itself. So you might have to resort to `onnx` python utility. Here is the code that can generate one:

```python
import onnx
import onnx.helper
import onnx.checker
import numpy as np

# Create input tensors
input1 = onnx.helper.make_tensor_value_info('input1', onnx.TensorProto.FLOAT, [3])
input2 = onnx.helper.make_tensor_value_info('input2', onnx.TensorProto.FLOAT, [3])
input3 = onnx.helper.make_tensor_value_info('input3', onnx.TensorProto.FLOAT, [3])

# Create output tensor
output = onnx.helper.make_tensor_value_info('output', onnx.TensorProto.FLOAT, [3])

# Create the Sum node
sum_node = onnx.helper.make_node(
    'Sum',
    inputs=['input1', 'input2', 'input3'],
    outputs=['output']
)

# Create the graph (GraphProto)
graph_def = onnx.helper.make_graph(
    nodes=[sum_node],
    name='SumGraph',
    inputs=[input1, input2, input3],
    outputs=[output]
)

# Create the model (ModelProto)
model_def = onnx.helper.make_model(graph_def, producer_name='sum-model')
onnx.checker.check_model(model_def)

# Save the ONNX model
onnx.save(model_def, 'sum_model.onnx')

print("ONNX model 'sum_model.onnx' generated successfully.")

```

However, for the underlying operator you'll have to use Burn's add operator multiple times. This is very similar to concat operation, see: https://github.com/tracel-ai/burn/blob/main/crates/burn-import/src/burn/node/concat.rs#L36-L38

Basically, a generated code might look like `input1` + `input2` + `input3`

**Comment by JachymPutta** [comment]
Ahh, I see, let me fix that!
Thank you!

File: crates/burn-import/src/burn/node/sum.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    use super::*;
+    use crate::burn::{
+        graph::BurnGraph,
+        node::{sum::SumNode, test::assert_tokens},
+        TensorType,
+    };
+
+    #[test]
+    fn test_codegen_concat() {
```
Comment (antimora) [inline crates/burn-import/src/burn/node/sum.rs:56]: test_codegen_concat => test_codegen_sum

**Review by antimora — APPROVED** [review]
There is one little rename required but overall looks great! 

Thank you for your contribution! You are on a roll!

Approving in advance.

File: crates/burn-import/src/burn/node/sum.rs (hunk 1 above)
Comment (JachymPutta) [reply crates/burn-import/src/burn/node/sum.rs:56]: Oh, I missed that! That's what I get for the mindless copying:)

**Comment by JachymPutta** [comment]
Should be good!
